### ISSUE TITLE
m.css crashes with AttributeError on Doxygen v1.9.1

### ISSUE THREAD

**Issue by marcizhu** [mosra/m.css#191]
For some reason, when using doxygen v1.9.X the script `m.css/documentation/doxygen.py` crashes with the following traceback:

```
Traceback (most recent call last):
  File "m.css/documentation/doxygen.py", line 3859, in <module>
    run(state, templates=os.path.abspath(args.templates), wildcard=args.wildcard, index_pages=args.index_pages, search_merge_subtrees=not args.search_no_subtree_merging, search_add_lookahead_barriers=not args.search_no_lookahead_barriers, search_merge_prefixes=not args.search_no_prefix_merging)
  File "m.css/documentation/doxygen.py", line 3716, in run
    parsed = parse_xml(state, file)
  File "m.css/documentation/doxygen.py", line 2988, in parse_xml
    func = parse_func(state, memberdef)
  File "m.css/documentation/doxygen.py", line 1965, in parse_func
    if signature.endswith('=default'):
AttributeError: 'NoneType' object has no attribute 'endswith'
```

As far as I know, this didn't happen with previous versions of Doxygen. I'm using m.css' latest commit (42d4a9a).

**Comment by crisluengo** [comment]
This issue is fixed here:
https://github.com/mosra/m.css/pull/189

**Comment by marcizhu** [comment]
@crisluengo I have tested your PR and I can confirm that the changes proposed on #189 do indeed fix this issue.
@mosra Can we get that PR merged?

**Comment by marzer** [comment]
`*bump*`

@mosra I've just hit this bug too :(

**Comment by GiulioRomualdi** [comment]
I had the same problem

**Comment by marcizhu** [comment]
@mosra Will this ever get fixed? It's been over four months already.

**Comment by mosra** [comment]
Sorry sorry extremely sorry, I'm on this now. Finally was able to drag through the bug swamp that piled up over the past two years and now onto the final boss, Doxygen 1.9.

Or actually, it can wait one more day, right? Need some sleep.

**Comment by mosra** [comment]
#189 is merged as c7ee89e74cf96323b7bf14d4c7535b6867d57868, so things should no longer blow up. Until another Doxygen release, at least.

**Comment by mosra** [comment]
FYI for whoever is subscribed to this issue: #215 tracks remaining issues for proper Doxygen 1.9 support. Some of them are rather nasty (like the inline namespace regression), so until these boxes are all ticked off I don't recommend updating.
